nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ConvertJSONtoCSV.java

```diff
@@ -0,0 +1,360 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.processors.standard;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.nifi.annotation.behavior.InputRequirement;
+import org.apache.nifi.annotation.behavior.WritesAttribute;
+import org.apache.nifi.annotation.documentation.CapabilityDescription;
+import org.apache.nifi.annotation.documentation.Tags;
+import org.apache.nifi.annotation.lifecycle.OnScheduled;
+import org.apache.nifi.components.AllowableValue;
+import org.apache.nifi.components.PropertyDescriptor;
+import org.apache.nifi.flowfile.FlowFile;
+import org.apache.nifi.processor.AbstractProcessor;
+import org.apache.nifi.processor.ProcessContext;
+import org.apache.nifi.processor.ProcessSession;
+import org.apache.nifi.processor.ProcessorInitializationContext;
+import org.apache.nifi.processor.Relationship;
+import org.apache.nifi.processor.exception.ProcessException;
+import org.apache.nifi.processor.io.StreamCallback;
+import org.apache.nifi.processor.util.StandardValidators;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.json.JSONTokener;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.SortedSet;
+import java.util.TreeSet;
+
+@Tags({"JSON, CSV, convert"})
+@InputRequirement(InputRequirement.Requirement.INPUT_REQUIRED)
+@WritesAttribute(attribute="mime.type", description="Sets the mime type to application/csv")
+@CapabilityDescription("Converts a JSON document to CSV. This processor reads the entire content " +
+        "of incoming FlowFiles into memory in order to perform the conversion. The processor will parse JSON Arrays, JSON Objects " +
+        "and the combination of the two regardless of the level of nesting in the JSON document.")
+public class ConvertJSONtoCSV extends AbstractProcessor {
+    volatile String delimiter;
+    volatile String removeFields;
+    volatile String emptyFields = "";
+
+    public static final AllowableValue INCLUDE_HEADER_TRUE = new AllowableValue(
+            "True", "True", "Creates headers for each JSON file.");
+    public static final AllowableValue INCLUDE_HEADER_FALSE = new AllowableValue(
+            "False", "False", "Only parses the JSON fields and does not include headers");
+
+
+    public static final PropertyDescriptor DELIMITER = new PropertyDescriptor
+            .Builder().name("CSV Delimiter")
```

Comment:
The latest convention is to include a name() that is language independent, with displayName() that can be more user-focusing. This is just a suggestion that will help with internationalization later.